chore: Reorder/fix data lineage based on latest changes (#1446)
seqeralabs/docs: platform-cloud/docs/data/data-lineage.md

---

:::info

> Data lineage in Platform is in public preview. Requires at least Nextflow `v25.04`, AWS S3 object storage, and Amazon Simple Queue Service (SQS). Best results with Nextflow `v26.04` and higher.

Reviewer: ```suggestion
Data lineage in Platform is currently in public preview. It requires Nextflow 25.04 or later, AWS S3 object storage, and Amazon Simple Queue Service (SQS). For best results, use Nextflow 26.04 or higher.
```

## How data lineage works

> Nextflow creates a structured JSON record for each entity in your pipeline during the execution of the workflow when lineage is enabled:

Reviewer: ```suggestion
When lineage is enabled, Nextflow generates a structured JSON record for each entity in your pipeline during workflow execution:
```

Each record gets a lineage ID (LID), a `lid://` URI that uniquely identifies the entity. Every LID and lineage label renders as a clickable link, and you can navigate to all related entities across your organization.

### Functional flow

1. Nextflow appends lineage record objects (`*.data.json`) to the defined object storage bucket.
1. The bucket is configured to filter for objects matching `.data.json` and sends object store notifications to the queue.

> 1. SQS queue receives `s3:ObjectCreated:*` events

Reviewer: ```suggestion
1. SQS queue receives `s3:ObjectCreated:*` events.
```

1. Platform reads the queue, returning the lineage objects created, and indexes them in the database.
1. The index enriches the [run details][run-details]
1. The index enriches the display of workflow-generated objects in Data Explorer with links to the origin pipeline run and task, sources of the object, and any lineage labels associated with the object.

## Enable data lineage

To start collecting data lineage for all pipeline runs in your workspace, go to **Settings > Workspace Settings**. Select **Lineage**. Toggle the **Enable lineage by default** on to collect data lineage for all pipeline runs in the workspace or toggle off to require per pipeline launch configuration. Choose either a **Manual** or an **Automatic** configuration for lineage resources: 

- **Manual**: Define the credentials, region, object storage bucket and path, and SQS queue ARN.
- **Automatic**: Define the credentials and region. The object storage bucket and path, and SQS queue, is managed by Platform.

> Once set and enabled, all pipeline runs in the workspace generate data lineage. See [Lineage][workspace-lineage] for more information about the settings. An indexer runs continuously in the background per workspace, long-polling the SQS queue for S3 event notifications. Failed event messages are left on the queue so they can be retried after the visibility timeout.

Reviewer: ```suggestion
Once set and enabled, all pipeline runs in the workspace generate data lineage. See [Lineage][workspace-lineage] for more information about the settings. An indexer runs continuously in the background per workspace, long-polling the SQS queue for S3 event notifications. Failed event messages remain in the queue and can be retried after the visibility timeout.
```
Please check meaning is retained.

### Lineage labels

Assign lineage labels to output files using the `label` directive in your Nextflow process definitions. Labels appear in lineage records.

Both Platform labels and Nextflow lineage labels propagate to lineage records. Platform excludes resource labels as they relate to underlying compute resources, not the data itself.

:::info
Nextflow lineage labels are **immutable**. They are set at execution time and cannot be changed. Platform labels are _mutable_ by design and can be changed after a run launches. Changing Platform labels post-launch will produce a mismatch between Platform run labels and Nextflow lineage labels.
:::

### Changing or disabling data lineage

If data lineage is **changed** from automatically-provisioned to manually-provisioned:

- **New object storage bucket**: Bucket notification rule is cleared, Platform-managed SQS queue is deleted. Events may be missed. Bucket and data are preserved.
- **Same object storage bucket, different SQS queue**: Redirect bucket notification rule to new SQS queue ARN, old Platform-managed SQS queue is deleted. Events may be missed. Bucket and data are preserved.

> - **Same object storage bucket, same SQS queue**: No changes to cloud provider resources. All events preserved. Bucket and data are preserved.

Reviewer: ```suggestion
- **New object storage bucket**: The bucket notification rule is cleared and the Platform-managed SQS queue is deleted. Some events may be missed. The bucket and its data are preserved.
- **Same object storage bucket, different SQS queue**: The bucket notification rule is redirected to the new SQS queue ARN, and the old Platform-managed SQS queue is deleted. Some events may be missed. The bucket and its data are preserved.
- **Same object storage bucket, same SQS queue**: No cloud provider resources change. All events, the bucket, and its data are preserved.
```
